whitebox/analytics/models/pipelines.py

```diff
@@ -138,3 +139,59 @@ def create_multiclass_classification_training_model_pipeline(
     multi_evaluation_report["precision"] = prec_score
 
     return clf, multi_evaluation_report
+
+
+def create_regression_training_model_pipeline(
+    training_dataset: pd.DataFrame, target: str, model_id: str
+) -> Dict[str, float]:
```

Comment:
I have some questions:
- why did we go with a model dependant approach on this? Its just a regression, right? We could do a simpler implementation.
- Even if we go with this implementation, training a model for regression requires a lot of params. Without them, it will just be a random prediction. I understand it is a level 0 feature importance but what is the value we provide with this approach?

Reply:
> I have some questions:
> 
> * why did we go with a model dependant approach on this? Its just a regression, right? We could do a simpler implementation.
> * Even if we go with this implementation, training a model for regression requires a lot of params. Without them, it will just be a random prediction. I understand it is a level 0 feature importance but what is the value we provide with this approach?

To be honest I can't totally understand both questions but I will try to answer them based on my understanding:
- What do you mean model dependent? The implementation is the simpler we can do taking into consideration that we don't have to insert any new library there. And what does it mean simple regression? Nothing is simple when we talk for models..... :) 
- Training a model for regression requires a lot of params, as a lot of params require any kind of model. Exactly, it is a random prediction...actually is as random as the classification models we have above. We have discussed it several times, but let me repeat it once again here. Without fine tuning, the possibilities of producing a garbage model is pretty high with whatever model we would use. 
cc: @stavrostheocharis

Reply:
Give me some time to think about this more. I understand what youre saying.
On the first bullet i meant that regression is self explanable.